src/control-plane/billing/billing-provider.ts

```diff
@@ -84,4 +132,26 @@ export class BillingProvider extends Construct {
       );
     }
   }
+
+  private getFunctionProps(
+    fn: IFunction | IFunctionTrigger,
+    defaultTrigger: DetailType
+  ): IFunctionTrigger {
+    return 'handler' in fn
+      ? { handler: fn.handler, trigger: fn.trigger }
+      : { handler: fn, trigger: defaultTrigger };
+  }
+
+  private createEventTarget(
+    eventManager: EventManager,
+    defaultEvent: DetailType,
+    fn?: IFunction | IFunctionTrigger
+  ) {
+    if (!fn) {
+      return;
+    }
+
```

Comment:
Does this fn parameter be optional? and why do we need this if condition?
I think fn should be mandatory and we should remove the if loop.

Reply:
This is just a helper function to avoid having to do checks for each of the many optional functions we have for the billing interface (like the `createUserFunction`).